app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,84 @@
+class RentalsController < ApplicationController
+  def index
+    rentals = Rental.all
+    render json: rentals.as_json(),
+           status: :ok
+  end
+
+  def check_out
+    customer = Customer.find_by(id: params[:customer_id])
+    video = Video.find_by(id: params[:video_id])
+
+    if video.nil? || customer.nil?
+      render json: {
+        errors: ["Not Found"],
+      }, status: :not_found
+      return
+    end
+    if !video.available
+      render json: {
+               ok: false,
+               errors: "Unable to create rental",
+             }, status: :bad_request
+      return
+    end
+
+    rental = Rental.new(customer_id: customer.id, video_id: video.id)
+    rental.check_out_date = Date.today
+    rental.due_date = Date.today + 7
+
+    if rental.save
+      video.decrease_inventory
+      customer.add_to_checked_out
+      render json: { customer_id: rental.customer_id,
+                     video_id: rental.video_id,
+                     due_date: rental.due_date,
+                     videos_checked_out_count: customer.videos_checked_out_count,
+                     available_inventory: video.available_inventory }, status: :ok
+      return
+    else
+      render json: {
+        ok: false,
+        errors: rental.errors.messages,
+      }, status: :bad_request
```

Comment:
There's actually no chance for the user input to be invalid in this case because we've already validated that we can find the customer and video specified by the user.
If we are going to add an `else` block here, we should actually return an `:internal_server_error` because the failure to save would be due to something wrong with the database.

Reply:
Please reach out to me if this comment is unclear! :)